test-basic/transactions.js

```diff
@@ -226,5 +228,43 @@ describe('transaction', function(){
         })
       .catch(done);
     });
+
+    it('should read the same file as written when using mlutil.convertTransaction', function(done){
+      var txn = null;
+      var tid = null;
+      var cookies = null;
+      db.transactions.open(true)
+      .result(function(response) {
+        txn = response;
+        return db.documents.write({
+          txid: txn,
+          uri: uri,
+          contentType: 'application/json',
+          content: {txKey: txn.txid}
+          }).result();
+        })
+      .then(function(response) {
+        tid = txn.txid;
+        cookies = txn.cookies[0];
+        var convertedTransaction = mlutil.convertTransaction(txn);
+        
+        return db.transactions.read(convertedTransaction).result() ;
+        })
+      .then(function(document) {
+        var status = document['transaction-status'];
+        var transactionId = status['transaction-id'];
+        var host = status['host'];
+        var hostId = 'HostId='+host['host-id'];
+
+        tid.should.equal(transactionId);
+        cookies.should.equal(hostId);
```

Comment:
Given that the sequence of cookies is under the control of the load balancer, it might be sufficient to skip the cookie inspection and just verify that the read request in the transaction succeeded and returned the same transaction id.